packages/valory/skills/decision_maker_abci/behaviours/purchase_subscription.py

```diff
@@ -0,0 +1,233 @@
+#
+# ------------------------------------------------------------------------------
+
+"""This module contains the behaviour for the decision-making of the skill."""
+import json
+from typing import Any, Generator, Optional, Dict
+
+from hexbytes import HexBytes
+
+from packages.valory.contracts.erc20.contract import ERC20
+from packages.valory.contracts.transfer_nft_condition.contract import TransferNftCondition
+from packages.valory.protocols.contract_api import ContractApiMessage
+from packages.valory.skills.decision_maker_abci.behaviours.base import (
+    DecisionMakerBaseBehaviour,
+)
+from packages.valory.skills.decision_maker_abci.models import MultisendBatch
+from packages.valory.skills.decision_maker_abci.payloads import DecisionReceivePayload, SubscriptionPayload
+from packages.valory.skills.decision_maker_abci.states.purchase_subscription import SubscriptionRound
+
+
+class PurchaseSubscriptionBehaviour(DecisionMakerBaseBehaviour):
+    """A behaviour in which the agents purchase a subscriptions."""
+
+    matching_round = SubscriptionRound
+
+    def __init__(self, **kwargs: Any) -> None:
+        """Initialize `RedeemBehaviour`."""
+        super().__init__(**kwargs)
+        self.purchase_tx: str = ""
+        self.approval_tx: str = ""
+
+    def _get_purchase_params(self) -> Generator[None, None, Optional[Dict[str, Any]]]:
+        """Get purchase params."""
+        # this data was taken from the nervermined api, by analyzing the following subscription
+        # https://gnosis.nevermined.app/en/subscription/did:nv:416e35cb209ecbfbf23e1192557b06e94c5d9a9afb025cce2e9baff23e907195
+        # more specifically the subscription creation tx: https://gnosisscan.io/tx/0x4cac38e6e28992ca8c20c92ffb8d0564dc7d1c3ca9f6b08db0a04b5f5fb23631
+        # and the purchase tx: https://gnosisscan.io/tx/0x086e693fac1c3d084dffdc899adaca773ad986e6e2bff5700549e1ae8364731e (this is what the service should do)
+
+        purchase_params = {}
+
+        mech = self.params.mech_agent_address
+        params = self.params.mech_to_subscription_params[mech]
+
+        base_url = params["base_url"]
+        did = params["did"]
+        purchase_params["did"] = did
+        purchase_params["nft_receiver"] = self.synchronized_data.safe_contract_address
+        purchase_params["contract_address"] = params["transfer_nft_condition_address"]
+
+        did_url = f"{base_url}/{did}"
+        response = yield from self.get_http_response(
+            method="GET",
+            url=did_url,
+            headers={"accept": "application/json"},
+        )
+        if response.status_code != 200:
+            self.context.logger.error(
+                f"Could not retrieve data from did url {did_url}. "
+                f"Received status code {response.status_code}."
+            )
+            return None
+
+        try:
+            # TODO: would be good to get a sanity check here, whether its possible to get this information from somewhere else
```

Comment:
would be good to get a sanity check on the this

Reply:
@aaitor it would be good if you or someone from your team can confirm that this logic is sensible